OpenSim/Simulation/Model/ContactGeometry.cpp

```diff
@@ -81,90 +74,110 @@ void ContactGeometry::constructProperties()
     constructProperty_color(defaultColor);
 }
 
-const Vec3& ContactGeometry::getLocation() const
+void ContactGeometry::constructConnectors()
 {
-    return get_location();
+    constructConnector<PhysicalFrame>("frame");
 }
 
+const Vec3& ContactGeometry::getLocation() const
+{ return get_location(); }
+
 void ContactGeometry::setLocation(const Vec3& location)
-{
-    set_location(location);
-}
+{ set_location(location); }
 
 const Vec3& ContactGeometry::getOrientation() const
-{
-    return get_orientation();
-}
+{ return get_orientation(); }
 
 void ContactGeometry::setOrientation(const Vec3& orientation)
-{
-    set_orientation(orientation);
-}
+{ set_orientation(orientation); }
 
-SimTK::Transform ContactGeometry::getTransform()
+SimTK::Transform ContactGeometry::findTransformInBaseFrame() const
 {
-    return SimTK::Transform(Rotation(SimTK::BodyRotationSequence,
-        get_orientation()[0], SimTK::XAxis,
-        get_orientation()[1], SimTK::YAxis,
-        get_orientation()[2], SimTK::ZAxis), get_location());
+    // B: base
+    // F: frame specified in the connector for this object.
+    // P: the (imaginary) frame that is defined (relative to F) by the location
+    //    and orientation properties.
+    const auto& X_BF = getFrame().findTransformInBaseFrame();
+    SimTK::Transform X_FP(
+            SimTK::Rotation(SimTK::BodyRotationSequence,
+                get_orientation()[0], SimTK::XAxis,
+                get_orientation()[1], SimTK::YAxis,
+                get_orientation()[2], SimTK::ZAxis),
+            get_location());
+    return X_BF * X_FP;
 }
 
-PhysicalFrame& ContactGeometry::updBody()
-{
-    return *_body;
-}
+SimTK::Transform ContactGeometry::getTransform() const
+{ return findTransformInBaseFrame(); }
 
-const PhysicalFrame& ContactGeometry::getBody() const
+const PhysicalFrame& ContactGeometry::getFrame() const
 {
-    return *_body;
+    return getConnector<PhysicalFrame>("frame").getConnectee();
 }
 
-void ContactGeometry::setBody(PhysicalFrame& body)
+void ContactGeometry::setFrame(const PhysicalFrame& frame)
 {
-    _body = &body;
-    set_body_name(body.getName());
+    updConnector<PhysicalFrame>("frame").connect(frame);
 }
 
-const std::string& ContactGeometry::getBodyName()
+const std::string& ContactGeometry::getFrameName() const
```

Comment:
Having `get/setFrame` is minimally complete. Now that we can connect directly to the frame and we aren't setting string properties anymore, we don't need the name-based interface anymore. Now seems like a good time to remove these.